nncf/common/tensor_statistics/statistics.py

```diff
@@ -26,23 +26,25 @@ class TensorStatistic(ABC):
     TENSOR_STATISTIC_OUTPUT_KEY = "tensor_statistic_output"
 
     @staticmethod
-    def tensor_eq(tensor1: Tensor, tensor2: Tensor, rtol=1e-6) -> bool:
+    def tensor_eq(tensor1: Tensor, tensor2: Tensor, rtol: float = 1e-6) -> Any:
         return fns.allclose(tensor1, tensor2)
 
     @abstractmethod
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> Any:
         pass
 
 
 class MinMaxTensorStatistic(TensorStatistic):
     MIN_STAT = "min_values"
     MAX_STAT = "max_values"
 
-    def __init__(self, min_values, max_values):
+    def __init__(self, min_values: Tensor, max_values: Tensor):
         self.min_values = min_values
         self.max_values = max_values
 
-    def __eq__(self, other: "MinMaxTensorStatistic") -> bool:
+    def __eq__(self, other: Any) -> Any:
```

Comment:
When I included all the requested changes, the mypy check fails in lines 48, 70 and 122 with the error `error: Returning Any from function declared to return "bool"  [no-any-return]`

Reply:
That because `self.tensor_eq` returns Any, I missed that. Please make `self.tesnor_eq` return bool

Reply:
If it will be failing please consider updating type hints of nested functions. __eq__ should return bools only

Reply:
> If it will be failing please consider updating type hints of nested functions. **eq** should return bools only

Yea it is failing with `nncf\common\tensor_statistics\statistics.py:30: error: Returning Any from function declared to return "bool"  [no-any-return]`

Reply:
> That because `self.tensor_eq` returns Any, I missed that. Please make `self.tesnor_eq` return bool

I would like to confirm if I can do the following:

`def tensor_eq(tensor1: Tensor, tensor2: Tensor, rtol: float = 1e-6) -> bool:`
`return cast(bool, fns.allclose(tensor1, tensor2))`

Reply:
Please do that